### DISCUSSION TITLE
How to use _and in a url for rest api filter?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#15875] [Archive]
The doc gives as exemple `?filter[first_name][_eq]=Rijk`
But how to write  `?filter[city][_eq]="New York" and [sport][_eq]=tennis`

**Comment by br41nslug** [comment]
You could either write that as ```?filter[city][_eq]=New%20York&filter[sport][_eq]=tennis``` or in json format ```?filter={"city":{"_eq":"New York"},"sport":{"_eq":"tennis"}}``` (or `?filter={"_and":[{"city":{"_eq":"New York"}},{"sport":{"_eq":"tennis"}}]}`)

**Reply by Pachat** [reply]
Thank. It is what I tried but was getting an `INTERNAL_SERVER_ERROR`. 
As the message does not guide much, I had to be sure it did not come from a syntax error.
_The error was in fact a type error: filtering an UUID with an integer._

BTW, how do you write the OR condition in the way of `?filter[city][_eq]=New%20York` OR `filter[sport][_eq]=tennis`?
Do we have to use the json_format only?

**Reply by br41nslug** [reply]
This format `?filter[city][_eq]=New%20York` is getting translated to an object too so you could do an `or` like this `?filter[_or][][city][_eq]=test&filter[_or][][sport][_eq]=tennis` but that gets bulky and finicky very fast. It's a lot easier to stick with the json format as you don't have the translation step from url notation to json in between.
